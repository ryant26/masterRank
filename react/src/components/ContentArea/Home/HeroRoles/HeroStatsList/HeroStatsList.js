import React from 'react';
import {connect} from 'react-redux';

import PropTypes from 'prop-types';
import HeroImage from '../../../../HeroImage/HeroImage';
import HeroStatsCard from "../HeroStatsListItem/HeroStatsListItem";

const HeroStatsList =  ({hero, heroes}) => {

    const userHeroes = heroes.filter((userHero) => userHero.platformDisplayName === hero.platformDisplayName).sort((hero1, hero2) => {
        return hero1.priority > hero2.priority;
    });

    const wins = userHeroes.reduce((wins, hero) => {
        if (hero.stats && hero.stats.wins) {
            return wins + hero.stats.wins;
        }
        return wins;
    }, 0);

    const srNode = hero.skillRating ? <span className="sub-title"><b>{hero.skillRating}</b> SR</span> : '';
    const winsNode = wins ? <span className="sub-title"><b>{wins}</b> WINS</span> : '';

    const heroImages = userHeroes.map((hero, i) => <HeroImage key={i} heroName={hero.heroName}/>);

    let heroStatsCards = userHeroes.filter((hero) => hero.stats).map((hero, i) => <HeroStatsCard key={i} hero={hero}/>);

    if (!heroStatsCards.length) {
        heroStatsCards = <div className="flex justify-center sub-title">We have no stats on this player.</div>;
    }

    return (
      <div className="HeroStatsList">
        <div className="header">
          <div className="flex justify-between align-center">
              <div>
                  <h3>{hero.platformDisplayName}</h3>
                  {srNode}
                  {winsNode}
              </div>
              <div className="preferredHeroIcons">
                  {heroImages}
              </div>
          </div>
        </div>
        <div className="body">
            {heroStatsCards}
        </div>
        <div className="footer flex justify-end align-center">
            <div className="button-secondary flex align-center">
                <div className="button-content">
                    INVITE TO GROUP
                </div>
            </div>
        </div>
      </div>
    );
};

HeroStatsList.propTypes = {
    hero: PropTypes.object,
    heroes: PropTypes.array.isRequired
};

const mapStateToProps = (state) => {
    return {
        heroes: state.heroes
    };
};

export default connect(mapStateToProps)(HeroStatsList);