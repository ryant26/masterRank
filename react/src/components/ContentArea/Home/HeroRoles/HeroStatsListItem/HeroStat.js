import React from 'react';
import PropTypes from 'prop-types';

const HeroStat = ({stat, percentile, statLabel, statName}) => {
    const indicatorLevel = {
        'width': (percentile * 100) + '%',
    };

    let statNode;

    if (stat) {
        stat = stat.toFixed(0);
        statNode = <div><span className="stat">{stat}</span><span className="stat-label sub-title">{statLabel}</span></div>;
    } else {
        statNode = <span className="stat">-</span>;
    }

    return (
        <div className="HeroStat flex flex-column">
            {statNode}
            <div className="indicator">
                <div className="indicator-level" style={indicatorLevel}/>
            </div>
            <div className="sub-title">{statName.toUpperCase()}</div>
        </div>
    );
};

HeroStat.propTypes = {
    stat: PropTypes.number,
    percentile: PropTypes.number.isRequired,
    statLabel: PropTypes.string.isRequired,
    statName: PropTypes.string.isRequired
};

export default HeroStat;