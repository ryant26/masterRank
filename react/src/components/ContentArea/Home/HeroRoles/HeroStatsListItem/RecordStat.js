import React from 'react';
import PropTypes from 'prop-types';

const RecordStat = ({stat, statName}) => {
    return (
        <div className="RecordStat flex flex-column align-end">
            <div className="stat">{stat}</div>
            <div className="sub-title">{statName.toUpperCase()}</div>
        </div>
    );
};

RecordStat.propTypes = {
    stat: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
    statName: PropTypes.string.isRequired
};


export default RecordStat;