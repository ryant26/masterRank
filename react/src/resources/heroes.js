const HEROES = [
    {
        "platformDisplayName": "PwNShoPP#1662",
        "heroName": "soldier76",
        "skillRating": 2400,
        "stats": {
            "hoursPlayed": 3,
            "wins": 11,
            "losses": 7,
            "gamesPlayed": 18,
            "kdRatio": 2.9936305732484074,
            "accuracy": 36,
            "blockedPerMin": 0,
            "healingPerMin": 148.0888888888889,
            "damagePerMin": 1338.638888888889,
            "avgObjElims": 10.5,
            "avgObjTime": 56.22222222222222,
            "pKdRatio": 1,
            "pAccuracy": 1,
            "pBlockedPerMin": 0,
            "pHealingPerMin": 1,
            "pDamagePerMin": 1,
            "pAvgObjElims": 1,
            "pAvgObjTime": 1
        }
    },
    {
        "platformDisplayName": "PwNShoPP#1662",
        "heroName": "hanzo",
        "skillRating": 2400,
        "stats": {
            "hoursPlayed": 3,
            "wins": 11,
            "losses": 7,
            "gamesPlayed": 18,
            "kdRatio": 2.9936305732484074,
            "accuracy": 36,
            "blockedPerMin": 0,
            "healingPerMin": 148.0888888888889,
            "damagePerMin": 1338.638888888889,
            "avgObjElims": 10.5,
            "avgObjTime": 56.22222222222222,
            "pKdRatio": 1,
            "pAccuracy": 1,
            "pBlockedPerMin": 0,
            "pHealingPerMin": 1,
            "pDamagePerMin": 1,
            "pAvgObjElims": 1,
            "pAvgObjTime": 1
        }
    },
    {
        "platformDisplayName": "PwNShoPP#1662",
        "heroName": "dva",
        "skillRating": 2400,
        "stats": {
            "hoursPlayed": 3,
            "wins": 11,
            "losses": 7,
            "gamesPlayed": 18,
            "kdRatio": 2.9936305732484074,
            "accuracy": 36,
            "blockedPerMin": 0,
            "healingPerMin": 148.0888888888889,
            "damagePerMin": 1338.638888888889,
            "avgObjElims": 10.5,
            "avgObjTime": 56.22222222222222,
            "pKdRatio": 1,
            "pAccuracy": 1,
            "pBlockedPerMin": 0,
            "pHealingPerMin": 1,
            "pDamagePerMin": 1,
            "pAvgObjElims": 1,
            "pAvgObjTime": 1
        }
    },
    {
        "platformDisplayName": "PwNShoPP#1662",
        "heroName": "zenyatta",
        "skillRating": 2400,
        "stats": {
            "hoursPlayed": 3,
            "wins": 11,
            "losses": 7,
            "gamesPlayed": 18,
            "kdRatio": 2.9936305732484074,
            "accuracy": 36,
            "blockedPerMin": 0,
            "healingPerMin": 148.0888888888889,
            "damagePerMin": 1338.638888888889,
            "avgObjElims": 10.5,
            "avgObjTime": 56.22222222222222,
            "pKdRatio": 1,
            "pAccuracy": 1,
            "pBlockedPerMin": 0,
            "pHealingPerMin": 1,
            "pDamagePerMin": 1,
            "pAvgObjElims": 1,
            "pAvgObjTime": 1
        }
    },
];

export default HEROES;
