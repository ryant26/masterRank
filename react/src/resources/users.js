export const users = [
    {
        "_id": 12,
        "platformDisplayName": "PwNShoPP#1662",
        "lastUpdated": "2017-10-15T01:43:36.459Z",
        "platform": "pc",
        "level": 155,
        "portrait": "https://d1u1mce87gyfbn.cloudfront.net/game/unlocks/0x025000000000115A.png",
        "region": "us"
    }
];