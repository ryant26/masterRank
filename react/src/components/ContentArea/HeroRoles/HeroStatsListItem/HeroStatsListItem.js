import React from 'react';
import HeroImage from "../../../HeroImage/HeroImage";
import PropTypes from 'prop-types';
import RecordStat from './RecordStat';
import HeroStat from './HeroStat';

const HeroStatsListItem = ({hero}) => {
    const statLabels = {
        perMinute: '/min',
        seconds: 'seconds',
        percent: '%'
    };

    const statNames = {
        damage: 'damage',
        healing: 'healing',
        blocked: 'blocked',
        objKills : 'obj. kills',
        objTime: 'obj. time',
        accuracy: 'accuracy',
        wins: 'wins',
        losses: 'losses',
        kdRatio: 'k/d'
    };

    return (
        <div className="HeroStatsListItem">
            <div className="flex align-center">
                <HeroImage heroName={hero.heroName}/>
                <div>
                    <h3>{hero.heroName[0].toUpperCase() + hero.heroName.slice(1)}</h3>
                    <div className="sub-title">{hero.stats.hoursPlayed} hours played</div>
                </div>
                <div className="flex justify-between record">
                    <RecordStat stat={hero.stats.wins} statName={statNames.wins}/>
                    <RecordStat stat={hero.stats.losses} statName={statNames.losses}/>
                    <RecordStat stat={hero.stats.kdRatio.toFixed(2)} statName={statNames.kdRatio}/>
                </div>
            </div>
            <div className="flex wrap justify-between">
                <HeroStat stat={hero.stats.damagePerMin}
                          percentile={hero.stats.pDamagePerMin} statLabel={statLabels.perMinute} statName={statNames.damage}/>
                <HeroStat stat={hero.stats.healingPerMin}
                          percentile={hero.stats.pHealingPerMin} statLabel={statLabels.perMinute} statName={statNames.healing}/>
                <HeroStat stat={hero.stats.blockedPerMin}
                          percentile={hero.stats.pBlockedPerMin} statLabel={statLabels.perMinute} statName={statNames.blocked}/>
                <HeroStat stat={hero.stats.avgObjElims}
                          percentile={hero.stats.pAvgObjElims} statLabel={statLabels.perMinute} statName={statNames.objKills}/>
                <HeroStat stat={hero.stats.avgObjTime}
                          percentile={hero.stats.pAvgObjTime} statLabel={statLabels.perMinute} statName={statNames.objTime}/>
                <HeroStat stat={hero.stats.accuracy}
                          percentile={hero.stats.pAccuracy} statLabel={statLabels.percent} statName={statNames.accuracy}/>
            </div>
        </div>
    );
};

HeroStatsListItem.propTypes = {
    hero: PropTypes.shape({
        heroName: PropTypes.string.isRequired,
        stats: PropTypes.object.isRequired
    }).isRequired
};

export default HeroStatsListItem;