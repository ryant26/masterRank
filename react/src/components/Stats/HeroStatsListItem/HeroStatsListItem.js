import React from 'react';
import HeroImage from "../../HeroImage/HeroImage";
import PropTypes from 'prop-types';
import RecordStat from './RecordStat';
import HeroStat from './HeroStat';

const HeroStatsListItem = ({hero}) => {
    const statLabels = {
        perMinute: '/min',
        seconds: 'seconds',
        percent: '%'
    };

    const statNames = {
        damage: 'damage',
        healing: 'healing',
        blocked: 'blocked',
        objKills : 'obj. kills',
        objTime: 'obj. time',
        accuracy: 'accuracy',
        wins: 'wins',
        losses: 'losses',
        kdRatio: 'k/d'
    };

    return (
        <div className="HeroStatsListItem">
            <div className="flex align-center">
                <HeroImage heroName={hero.heroName}/>
                <div>
                    <h3>{hero.heroName[0].toUpperCase() + hero.heroName.slice(1)}</h3>
                    <div className="sub-title">{hero.stats.hoursPlayed} hours played</div>
                </div>
                <div className="flex justify-between record">
                    <RecordStat stat={hero.stats.wins || 0} statName={statNames.wins}/>
                    <RecordStat stat={hero.stats.losses || 0} statName={statNames.losses}/>
                    <RecordStat stat={(hero.stats.kdRatio ? hero.stats.kdRatio : 0).toFixed(2)} statName={statNames.kdRatio}/>
                </div>
            </div>
            <div className="flex wrap justify-between">
                <HeroStat stat={hero.stats.damagePerMin || 0}
                          percentile={hero.stats.pDamagePerMin} statLabel={statLabels.perMinute} statName={statNames.damage}/>
                <HeroStat stat={hero.stats.healingPerMin || 0}
                          percentile={hero.stats.pHealingPerMin} statLabel={statLabels.perMinute} statName={statNames.healing}/>
                <HeroStat stat={hero.stats.blockedPerMin || 0}
                          percentile={hero.stats.pBlockedPerMin} statLabel={statLabels.perMinute} statName={statNames.blocked}/>
                <HeroStat stat={hero.stats.avgObjElims || 0}
                          percentile={hero.stats.pAvgObjElims} statLabel={statLabels.perMinute} statName={statNames.objKills}/>
                <HeroStat stat={hero.stats.avgObjTime || 0}
                          percentile={hero.stats.pAvgObjTime} statLabel={statLabels.perMinute} statName={statNames.objTime}/>
                <HeroStat stat={hero.stats.accuracy || 0}
                          percentile={hero.stats.pAccuracy} statLabel={statLabels.percent} statName={statNames.accuracy}/>
            </div>
        </div>
    );
};

HeroStatsListItem.propTypes = {
    hero: PropTypes.shape({
        heroName: PropTypes.string.isRequired,
        stats: PropTypes.object.isRequired
    }).isRequired
};

export default HeroStatsListItem;