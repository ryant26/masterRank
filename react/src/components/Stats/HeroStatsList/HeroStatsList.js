import React from 'react';
import PropTypes from 'prop-types';
import HeroStatsListItem from "../HeroStatsListItem/HeroStatsListItem";

const HeroStatsList =  ({heroes, emptyMessage, showPlatformDisplayName, groupLeader}) => {

    let heroStatsCards = heroes.filter((hero) => hero.stats).map((hero, i) =>
        (
            <HeroStatsListItem
                key={i}
                hero={hero}
                isLeader={groupLeader === hero.platformDisplayName}
                showPlatformDisplayName={showPlatformDisplayName}
            />
        ));

    if (!heroStatsCards.length) {
        heroStatsCards = <div className="flex justify-center sub-title">{emptyMessage}</div>;
    }

    return (
      <div className="HeroStatsList">
        {heroStatsCards}
      </div>
    );
};

HeroStatsList.propTypes = {
    heroes: PropTypes.array.isRequired,
    emptyMessage: PropTypes.string,
    showPlatformDisplayName: PropTypes.bool,
    groupLeader: PropTypes.string
};

HeroStatsList.defaultProps = {
    emptyMessage: 'We have no stats on this player.',
    showPlatformDisplayName: false,
    groupLeader: ''
};

export default HeroStatsList;