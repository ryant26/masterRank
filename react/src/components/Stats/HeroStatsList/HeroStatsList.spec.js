import React from 'react';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import HeroStatsList from './HeroStatsList';
import HeroStatsListItem from 'components/Stats/HeroStatsListItem/HeroStatsListItem';
import { users } from 'resources/users';
import { getHeroes } from 'resources/heroes';

const mockStore = configureStore();
const shallowHeroStatsList = (heroes, groupLeader='', showPlatformDisplayName=false, isPending=false) => {
    return shallow(
        <HeroStatsList
            heroes={heroes}
            groupLeader={groupLeader}
            showPlatformDisplayName={showPlatformDisplayName}
            isPending={isPending}
        />
    );
};

describe('HeroStatsList Component', () => {
    const heroes = getHeroes();
    const groupLeader = heroes[0].platformDisplayName;
    let wrapper;

    beforeEach(() => {
        wrapper = shallowHeroStatsList(heroes);
    });

    it('should match the snapshot', () => {
        let store = mockStore({
            user: users[0]
        });
        let component = renderer.create(
            <Provider store={store}>
                <HeroStatsList heroes={getHeroes()}/>
            </Provider>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render without exploding', () => {
        expect(wrapper.find(HeroStatsList)).toBeTruthy();
    });

    it('should have a hero card for every hero passed', () => {
        expect(wrapper.find(HeroStatsListItem)).toHaveLength(4);
    });

    it('should show the heroStatsListItems in the order of preference', () => {
        let statsCards = wrapper.find(HeroStatsListItem);
        expect(statsCards.length).toBeGreaterThan(0);
        statsCards.forEach((image, i) => {
            let cardHeroName = image.props().hero.heroName;
            let heroName = heroes[i].heroName;

            expect(cardHeroName).toEqual(heroName);
        });
    });

    it('should set showPlatformDisplayName prop to false by default', () => {
        expect(wrapper.find(HeroStatsListItem).first().props().showPlatformDisplayName).toBeFalsy();
    });

    it('should set showPlatformDisplayName prop to true when passed in as true', () => {
        wrapper = shallowHeroStatsList(heroes, groupLeader, true);
        expect(wrapper.find(HeroStatsListItem).first().props().showPlatformDisplayName).toBeTruthy();
    });

    it('should set isLeader prop to true when hero belongs to group leader', () => {
        wrapper = shallowHeroStatsList(heroes, groupLeader);
        expect(wrapper.find(HeroStatsListItem).first().props().hero.platformDisplayName).toBe(groupLeader);
        expect(wrapper.find(HeroStatsListItem).first().props().isLeader).toBe(true);
    });

    it('should set isLeader prop to false when hero does not belongs to group leader', () => {
        expect(wrapper.find(HeroStatsListItem).first().props().hero.platformDisplayNam).not.toBe(heroes[0].platformDisplayName);
        expect(wrapper.find(HeroStatsListItem).first().props().isLeader).toBe(false);
    });

    it('should set hero stats list item isPending prop to false by default', () => {
        expect(wrapper.find(HeroStatsListItem).first().props().isPending).toBeFalsy();
    });

    it('should set hero stats list item isPending prop to true when true is given', () => {
        let isPending = true;
        wrapper = shallowHeroStatsList(heroes, groupLeader, false, isPending);
        expect(wrapper.find(HeroStatsListItem).first().props().isPending).toBeTruthy();
    });
});
