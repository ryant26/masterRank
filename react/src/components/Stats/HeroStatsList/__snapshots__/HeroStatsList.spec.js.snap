// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroStatsList Component should match the snapshot 1`] = `
<div
  className="HeroStatsList"
>
  <div
    className="HeroStatsListItem"
  >
    <div
      className="flex align-center"
    >
      <div>
        <div
          className="DisableableHeroImage"
        >
          <img
            alt="soldier76 icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div>
        <h3>
          
          Soldier76
        </h3>
        <div
          className="sub-title"
        >
          3 hours played
        </div>
      </div>
      <div
        className="flex justify-between record"
      >
        <div
          className="RecordStat flex flex-column align-end"
        >
          <div
            className="stat"
          >
            11
          </div>
          <div
            className="sub-title"
          >
            WINS
          </div>
        </div>
        <div
          className="RecordStat flex flex-column align-end"
        >
          <div
            className="stat"
          >
            7
          </div>
          <div
            className="sub-title"
          >
            LOSSES
          </div>
        </div>
        <div
          className="RecordStat flex flex-column align-end"
        >
          <div
            className="stat"
          >
            2.99
          </div>
          <div
            className="sub-title"
          >
            K/D
          </div>
        </div>
      </div>
    </div>
    <div
      className="flex wrap justify-between"
    >
      <div
        className="HeroStat flex flex-column"
      >
        <div>
          <span
            className="stat"
          >
            1339
          </span>
          <span
            className="stat-label sub-title"
          >
            /min
          </span>
        </div>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          DAMAGE
        </div>
      </div>
      <div
        className="HeroStat flex flex-column"
      >
        <div>
          <span
            className="stat"
          >
            148
          </span>
          <span
            className="stat-label sub-title"
          >
            /min
          </span>
        </div>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          HEALING
        </div>
      </div>
      <div
        className="HeroStat flex flex-column"
      >
        <span
          className="stat"
        >
          -
        </span>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "0%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          BLOCKED
        </div>
      </div>
      <div
        className="HeroStat flex flex-column"
      >
        <div>
          <span
            className="stat"
          >
            11
          </span>
          <span
            className="stat-label sub-title"
          >
            /min
          </span>
        </div>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          OBJ. KILLS
        </div>
      </div>
      <div
        className="HeroStat flex flex-column"
      >
        <div>
          <span
            className="stat"
          >
            56
          </span>
          <span
            className="stat-label sub-title"
          >
            /min
          </span>
        </div>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          OBJ. TIME
        </div>
      </div>
      <div
        className="HeroStat flex flex-column"
      >
        <div>
          <span
            className="stat"
          >
            36
          </span>
          <span
            className="stat-label sub-title"
          >
            %
          </span>
        </div>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          ACCURACY
        </div>
      </div>
    </div>
  </div>
  <div
    className="HeroStatsListItem"
  >
    <div
      className="flex align-center"
    >
      <div>
        <div
          className="DisableableHeroImage"
        >
          <img
            alt="hanzo icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div>
        <h3>
          
          Hanzo
        </h3>
        <div
          className="sub-title"
        >
          3 hours played
        </div>
      </div>
      <div
        className="flex justify-between record"
      >
        <div
          className="RecordStat flex flex-column align-end"
        >
          <div
            className="stat"
          >
            11
          </div>
          <div
            className="sub-title"
          >
            WINS
          </div>
        </div>
        <div
          className="RecordStat flex flex-column align-end"
        >
          <div
            className="stat"
          >
            7
          </div>
          <div
            className="sub-title"
          >
            LOSSES
          </div>
        </div>
        <div
          className="RecordStat flex flex-column align-end"
        >
          <div
            className="stat"
          >
            2.99
          </div>
          <div
            className="sub-title"
          >
            K/D
          </div>
        </div>
      </div>
    </div>
    <div
      className="flex wrap justify-between"
    >
      <div
        className="HeroStat flex flex-column"
      >
        <div>
          <span
            className="stat"
          >
            1339
          </span>
          <span
            className="stat-label sub-title"
          >
            /min
          </span>
        </div>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          DAMAGE
        </div>
      </div>
      <div
        className="HeroStat flex flex-column"
      >
        <div>
          <span
            className="stat"
          >
            148
          </span>
          <span
            className="stat-label sub-title"
          >
            /min
          </span>
        </div>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          HEALING
        </div>
      </div>
      <div
        className="HeroStat flex flex-column"
      >
        <span
          className="stat"
        >
          -
        </span>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "0%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          BLOCKED
        </div>
      </div>
      <div
        className="HeroStat flex flex-column"
      >
        <div>
          <span
            className="stat"
          >
            11
          </span>
          <span
            className="stat-label sub-title"
          >
            /min
          </span>
        </div>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          OBJ. KILLS
        </div>
      </div>
      <div
        className="HeroStat flex flex-column"
      >
        <div>
          <span
            className="stat"
          >
            56
          </span>
          <span
            className="stat-label sub-title"
          >
            /min
          </span>
        </div>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          OBJ. TIME
        </div>
      </div>
      <div
        className="HeroStat flex flex-column"
      >
        <div>
          <span
            className="stat"
          >
            36
          </span>
          <span
            className="stat-label sub-title"
          >
            %
          </span>
        </div>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          ACCURACY
        </div>
      </div>
    </div>
  </div>
  <div
    className="HeroStatsListItem"
  >
    <div
      className="flex align-center"
    >
      <div>
        <div
          className="DisableableHeroImage"
        >
          <img
            alt="dva icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div>
        <h3>
          
          Dva
        </h3>
        <div
          className="sub-title"
        >
          3 hours played
        </div>
      </div>
      <div
        className="flex justify-between record"
      >
        <div
          className="RecordStat flex flex-column align-end"
        >
          <div
            className="stat"
          >
            11
          </div>
          <div
            className="sub-title"
          >
            WINS
          </div>
        </div>
        <div
          className="RecordStat flex flex-column align-end"
        >
          <div
            className="stat"
          >
            7
          </div>
          <div
            className="sub-title"
          >
            LOSSES
          </div>
        </div>
        <div
          className="RecordStat flex flex-column align-end"
        >
          <div
            className="stat"
          >
            2.99
          </div>
          <div
            className="sub-title"
          >
            K/D
          </div>
        </div>
      </div>
    </div>
    <div
      className="flex wrap justify-between"
    >
      <div
        className="HeroStat flex flex-column"
      >
        <div>
          <span
            className="stat"
          >
            1339
          </span>
          <span
            className="stat-label sub-title"
          >
            /min
          </span>
        </div>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          DAMAGE
        </div>
      </div>
      <div
        className="HeroStat flex flex-column"
      >
        <div>
          <span
            className="stat"
          >
            148
          </span>
          <span
            className="stat-label sub-title"
          >
            /min
          </span>
        </div>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          HEALING
        </div>
      </div>
      <div
        className="HeroStat flex flex-column"
      >
        <span
          className="stat"
        >
          -
        </span>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "0%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          BLOCKED
        </div>
      </div>
      <div
        className="HeroStat flex flex-column"
      >
        <div>
          <span
            className="stat"
          >
            11
          </span>
          <span
            className="stat-label sub-title"
          >
            /min
          </span>
        </div>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          OBJ. KILLS
        </div>
      </div>
      <div
        className="HeroStat flex flex-column"
      >
        <div>
          <span
            className="stat"
          >
            56
          </span>
          <span
            className="stat-label sub-title"
          >
            /min
          </span>
        </div>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          OBJ. TIME
        </div>
      </div>
      <div
        className="HeroStat flex flex-column"
      >
        <div>
          <span
            className="stat"
          >
            36
          </span>
          <span
            className="stat-label sub-title"
          >
            %
          </span>
        </div>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          ACCURACY
        </div>
      </div>
    </div>
  </div>
  <div
    className="HeroStatsListItem"
  >
    <div
      className="flex align-center"
    >
      <div>
        <div
          className="DisableableHeroImage"
        >
          <img
            alt="zenyatta icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div>
        <h3>
          
          Zenyatta
        </h3>
        <div
          className="sub-title"
        >
          3 hours played
        </div>
      </div>
      <div
        className="flex justify-between record"
      >
        <div
          className="RecordStat flex flex-column align-end"
        >
          <div
            className="stat"
          >
            11
          </div>
          <div
            className="sub-title"
          >
            WINS
          </div>
        </div>
        <div
          className="RecordStat flex flex-column align-end"
        >
          <div
            className="stat"
          >
            7
          </div>
          <div
            className="sub-title"
          >
            LOSSES
          </div>
        </div>
        <div
          className="RecordStat flex flex-column align-end"
        >
          <div
            className="stat"
          >
            2.99
          </div>
          <div
            className="sub-title"
          >
            K/D
          </div>
        </div>
      </div>
    </div>
    <div
      className="flex wrap justify-between"
    >
      <div
        className="HeroStat flex flex-column"
      >
        <div>
          <span
            className="stat"
          >
            1339
          </span>
          <span
            className="stat-label sub-title"
          >
            /min
          </span>
        </div>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          DAMAGE
        </div>
      </div>
      <div
        className="HeroStat flex flex-column"
      >
        <div>
          <span
            className="stat"
          >
            148
          </span>
          <span
            className="stat-label sub-title"
          >
            /min
          </span>
        </div>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          HEALING
        </div>
      </div>
      <div
        className="HeroStat flex flex-column"
      >
        <span
          className="stat"
        >
          -
        </span>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "0%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          BLOCKED
        </div>
      </div>
      <div
        className="HeroStat flex flex-column"
      >
        <div>
          <span
            className="stat"
          >
            11
          </span>
          <span
            className="stat-label sub-title"
          >
            /min
          </span>
        </div>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          OBJ. KILLS
        </div>
      </div>
      <div
        className="HeroStat flex flex-column"
      >
        <div>
          <span
            className="stat"
          >
            56
          </span>
          <span
            className="stat-label sub-title"
          >
            /min
          </span>
        </div>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          OBJ. TIME
        </div>
      </div>
      <div
        className="HeroStat flex flex-column"
      >
        <div>
          <span
            className="stat"
          >
            36
          </span>
          <span
            className="stat-label sub-title"
          >
            %
          </span>
        </div>
        <div
          className="indicator"
        >
          <div
            className="indicator-level"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="sub-title"
        >
          ACCURACY
        </div>
      </div>
    </div>
  </div>
</div>
`;
